source/Octopus.Data/Resources/Attributes/PropertyApplicabilityAttribute.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+
+namespace Octopus.Data.Resources.Attributes
+{
+    /// <summary>
+    /// Use this attribute to specify whether a given property is applicable based on the value of another property
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Property)]
+    public abstract class PropertyApplicabilityAttribute : Attribute
+    {
+        internal PropertyApplicabilityAttribute(PropertyApplicabilityMode mode, string propertyName, object propertyValue)
+        {
+            Mode = mode;
+            PropertyName = propertyName;
+            PropertyValue = propertyValue;
+        }
+        
+        public PropertyApplicabilityMode Mode { get; }
+        
+        /// <summary>
+        /// The name of the property that the property the attribute has been applied to depends on
```

Comment:
I had to read this `summary` 3 or 4 times before I got what it was trying to say...not sure how we can make it easier to grok though...lots of `the` and `property` in there :)

Reply:
Yea, confusing to the point where I don't really get what it's doing.

Would this be accurate:

_The name of the property that the other property (that the attribute has been applied to) depends on_ ?

Reply:
I've renamed the properties to remove confusion, and the need for the comment